app/uk/gov/hmrc/individualsdetailsapi/services/PathTree.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.individualsdetailsapi.service
+
+class PathTree(node: B) {
+
+  def listChildren: Iterable[String] = node.listChildren
+
+  def children: Iterable[Node] = node.children
+
+  def hasChild(key: String): Boolean = node.hasChild(key)
+
+  def getChild(key: String): Option[Node] = node.getChild(key)
+
+  override def toString: String =
+    children.toList.sortBy(c => c.get).map(c => c.toString).mkString(",")
+}
+
+object PathTree {
+
+  def apply(paths: Iterable[String], divider: String): PathTree = {
+    def pathToNode(p: Iterable[String]): Node =
+      if (p.size == 1) {
+        L(p.head)
+      } else {
+        B(p.head, Seq(pathToNode(p.tail)))
+      }
+
+    new PathTree(
+      B("root", Seq()) ++ paths
+        .map(path => path.split(divider))
+        .map(path => pathToNode(path)))
+  }
+
+  def apply(paths: Iterable[String]): PathTree =
+    PathTree(paths, "/")
+}
+
+sealed abstract class Node {
+
+  def get: String
+
+  def listChildren: Iterable[String] = List()
+
+  def hasChild(key: String): Boolean = false
+
+  def getChild(key: String): Option[Node] = None
+
+  private[service] def +(node: Node): B
+
+  private[service] def ++(nodes: Iterable[Node]): B
+}
+
+case class L(value: String) extends Node {
+
+  def get: String = this.value
+
+  private[service] def +(node: Node): B = B(value, Seq(node))
+
+  private[service] def ++(nodes: Iterable[Node]): B =
+    this + nodes.head ++ nodes.tail
+
+  override def toString: String = value
+}
+
+case class B(value: String, children: Iterable[Node]) extends Node {
```

Comment:
Could we have more meaningful names other than L and B?

Reply:
@parkeradam - this is a straight copy of @owen-opencast handler pattern from the other service. 

Could likely come up with something but would need to be uniform across all services.

Reply:
Ok, I'll resolve for now but think we may need to bash heads and come up with some better names at some point.